megameklab/src/megameklab/ui/MegaMekLabTabbedUI.java

```diff
@@ -0,0 +1,352 @@
+
+        if (CConfig.getStartUpType() == MMLStartUp.RESTORE_TABS) {
+            try {
+                TabStateUtil.saveTabState(editors.stream().limit(editors.size() - 1).toList());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        return true;
+    }
+
+    private void restrictToScrenSize() {
+        DisplayMode currentMonitor = getGraphicsConfiguration().getDevice().getDisplayMode();
+        int scaledMonitorW = UIUtil.getScaledScreenWidth(currentMonitor);
+        int scaledMonitorH = UIUtil.getScaledScreenHeight(currentMonitor);
+        int w = Math.min(getSize().width, scaledMonitorW);
+        int h = Math.min(getSize().height, scaledMonitorH);
+        setSize(new Dimension(w, h));
+    }
+
+
+    //<editor-fold desc="MenuBarOwner interface implementation">
+    @Override
+    public JFrame getFrame() {
+        return this;
+    }
+
+    @Override
+    public Entity getEntity() {
+        return currentEditor().getEntity();
+    }
+
+    @Override
+    public String getFileName() {
+        return currentEditor().getFileName();
+    }
+
+    @Override
+    public boolean hasEntityNameChanged() {
+        return currentEditor().hasEntityNameChanged();
+    }
+
+    @Override
+    public void refreshMenuBar() {
+        menuBar.refreshMenuBar();
+    }
+
+    @Override
+    public MenuBar getMMLMenuBar() {
+        return menuBar;
+    }
+    //</editor-fold>
+
+
+    /**
+     * Represents a button used for creating new tabs in the MegaMekLabTabbedUI interface.
+     * This class extends JPanel and is rendered as a non-opaque panel containing a "+" symbol.
+     * Used to mimic functionality for adding new tabs in a tabbed user interface.
+     * <p>
+     * The only reason this is a separate class instead of just the text for a tab is so that it can be green.
+     */
+    private static class NewTabButton extends JPanel {
+        public NewTabButton() {
+            setOpaque(false);
+            var label = new JLabel("➕");
+            label.setForeground(Color.GREEN);
+            add(label);
+        }
+    }
+
+    /**
+     * Represents a custom tab component for use in a tabbed user interface, designed to display
+     * the name of a unit and provide a close button for removing the associated tab.
+     * The close button can be shift-clicked to skip the editor's safety prompt.
+     * This class extends JPanel and is initialized with a unit name and its associated editor instance.
+     */
+    private class ClosableTab extends JPanel {
+        JLabel unitName;
+        JButton closeButton;
+        MegaMekLabMainUI editor;
+
+        public ClosableTab(String name, MegaMekLabMainUI mainUI) {
+            unitName = new JLabel(name);
+            editor = mainUI;
+
+            setOpaque(false);
+
+            closeButton = new JButton("❌");
+            closeButton.setForeground(Color.RED);
```

Comment:
Request that you add:
```
            closeButton.setFont(new Font("Symbola", Font.PLAIN, closeButton.getFont().getSize()));
 ```
so that non-Windows users can see the symbols.